mailbox/store/src/main/java/org/apache/james/mailbox/store/PreDeletionHooks.java

```diff
@@ -0,0 +1,50 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.store;
+
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.apache.james.mailbox.extension.PreDeletionHook;
+
+import com.google.common.collect.ImmutableSet;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class PreDeletionHooks {
+    public static PreDeletionHooks NO_PRE_DELETION_HOOK = new PreDeletionHooks(ImmutableSet.of());
+
+    private final Set<PreDeletionHook> hooks;
+
+    @Inject
+    public PreDeletionHooks(Set<PreDeletionHook> hooks) {
```

Comment:
why it's public if you use injection for it?

Reply:
Because some tests located in other packages also call this constructor

Reply:
So you can very well use the @VisibleForTesting annotation + public, at least it documents the situation